src/proxy/inbound.rs

```diff
@@ -221,7 +221,8 @@ impl Inbound {
 
         let rbac_ctx = crate::state::ProxyRbacContext {
             conn,
-            dest_workload_info: pi.proxy_workload_info.clone(),
+            dest_workload_info: Some(pi.local_workload_information.workload_info()),
```

Comment:
(musing/aside/future work) One thing I still don't like is that we are carrying these two things as state everywhere for a bunch of decisions, it still doesn't seem correct.

Carrying on from https://github.com/istio/ztunnel/pull/1218#discussion_r1722283016 - I don't think `WorkloadInfo` should ever be used to make RBAC decisions? `Workload`/`Identity` should be all we use, by the point we check RBAC we should have gotten all we need from WorkloadInfo and either converted it into an Identity or done whatever Identity-based checks we needed.

Reply:
Good news -- we literally do not use it at all so it was trivial to just remove it. Technically we could just get the policies + namespace (vs storing the whole workload), but i don't want to make a clone so its easier to just pass the WL around